add installation troubleshoot (#655)
run-ai/docs: docs/admin/runai-setup/cluster-setup/cluster-install.md

Below are instructions on how to install a Run:ai cluster.

## Prerequisites
Before installing, please review the installation prerequisites here: [Run:ai GPU Cluster Prerequisites](cluster-prerequisites.md).

!!! Important

> We strongly recommend running the Run:ai [pre-install script](cluster-prerequisites.md#pre-install-script) to verify that all prerequisites are met.

Reviewer: @roir This will not render as an admonition. The indentation is needed to tell markdown what to do.
```suggestion
    We strongly recommend running the Run:ai [pre-install script](cluster-prerequisites.md#pre-install-script) to verify that all prerequisites are met.
```

Using the cluster wizard:

* Choose a name for your cluster.
* Choose the Run:ai version for the cluster.
* Choose a target Kubernetes distribution (see [table](cluster-prerequisites.md#kubernetes) for supported distributions).

> * (SaaS and remote self-hosted cluster only) Enter a URL for the Kubernetes cluster. The URL need only be accessible within the organization's network. For more informtaion see [here](cluster-prerequisites.md#cluster-url).

Reviewer: @roir We generally do not use "here" but the name of the section. This is likely not your correction, but I fix these when I see them. 
```suggestion
* (SaaS and remote self-hosted cluster only) Enter a URL for the Kubernetes cluster. The URL need only be accessible within the organization's network. For more information, see [Cluster prerequisites](cluster-prerequisites.md#cluster-url).
```

      message: Dependencies are fulfilled
    - type: Deployed               
      status: "True"
      reason: deployed
      message: Resources Deployed
    - type: Available
      status: "True"
      reason: available
      message: System Available
    - type: Reconciled              # (2)
      status: "True"
      reason: reconciled
      message: Reconciliation completed successfully
  optional:  # (3)
    knative:    # (4)  
      components:
        hpa:
          available: true
        knative:
          available: true
        kourier:
          available: true
    mpi:        # (5) 
      available: true
```

1. Verifies that all mandatory dependencies are met: NVIDIA GPU Operator, Prometheus and NGINX controller.
2. Verifies that all of Run:ai managed resources have been successfully deployed.
3. Checks whether optional product dependencies have been met.
4. See [Inference prerequisites](cluster-prerequisites.md#inference).
5. See [distributed training prerequisites](cluster-prerequisites.md#distributed-training).

For a more extensive verification of cluster health, see [Determining the health of a cluster](../../troubleshooting/cluster-health-check.md).

> ### Troubleshooting
> #### Dependencies are not fulfilled

Reviewer: @roir 
```suggestion
### Troubleshooting

#### Dependencies are not fulfilled

```

1. Make sure to install the missing dependencies.
2. If dependencies are installed, make sure that the CRDs of said dependency are installed, and that the version is supported

> 3. Make sure there are no necessary adjustments for specific flavors as noted here https://docs.run.ai/v2.15/admin/runai-setup/cluster-setup/cluster-prerequisites

Reviewer: @roir
```suggestion
3. Make sure there are no necessary adjustments for specific flavors as noted in the [Cluster prerequisites](cluster-prerequisites.md)
```

#### Resources not deployed / System Unavailable / Reconciliation Failed

> 1. Run Preinstall diagnostic script https://docs.run.ai/v2.15/admin/runai-setup/cluster-setup/cluster-prerequisites/#pre-install-script and check for issues

Reviewer: @roir 
```suggestion
1. Run the [Preinstall diagnostic script](cluster-prerequisites.md#pre-install-script) and check for issues
```

> 2. Run

Reviewer: @roir 
```suggestion
2. Then run
```

Author: it's not a "then" relation between 1 and 2 they are separate suggestions

Reviewer: Then it would be an `or`?

Author: not necessarily. it can be an and

> ```
>    kubectl get pods -n runai
>    kubectl get pods -n monitoring
> ```
> and obtain logs from any failing pod `kubectl logs <pod_name>`

Reviewer: @roir This will render, but will create warnings and errors in the linter. There need to be spaces before and after codeblocks.
```suggestion

```
   kubectl get pods -n runai
   kubectl get pods -n monitoring
```

and obtain logs from any failing pod `kubectl logs <pod_name>`
```

> ```
>    kubectl get pods -n runai
>    kubectl get pods -n monitoring
> ```
> and obtain logs from any failing pod `kubectl logs <pod_name>`

Reviewer: @roir 
```suggestion

```
   kubectl get pods -n runai
   kubectl get pods -n monitoring
```

Then run `kubectl logs <pod_name>` to get logs from any failing pod.

> #### Common Issues
> 1. Run:ai was previously installed in the cluster and was deleted unsuccessfully, resulting in remaining CRDs
>     2. Diagnosis: This can be detected by running `kubectl get crds` in the relevant namespaces (or adding `-A` and checking for Run:ai CRDs)
>     3. Solution: Force delete said CRDs and reinstall
> 3. One or more of the pods have issues around valid certificates
>     4. Diagnosis: The logs contains a message similar to the following `failed to verify certificate: x509: certificate signed by unknown authority`
>     4. Solution:
>         5. This is usually due to an expired or invalid certificate in the cluster, and if so - renew the certificate
>         6. If Certificate is valid, but is Signed by a local CA, make sure to have followed the steps here https://docs.run.ai/v2.15/admin/runai-setup/config/org-cert/
> 

Reviewer: @roir I am going to nee to merge this PR to check some of the markdown. I don't have a good way of previewing all the indentations to ensure that it renders correctly.
```suggestion
#### Common Issues

1. Run:ai was previously installed in the cluster and was deleted unsuccessfully, resulting in remaining CRDs.
    1. Diagnosis: This can be detected by running `kubectl get crds` in the relevant namespaces (or adding `-A` and checking for Run:ai CRDs).
    2. Solution: Force delete the listed CRDs and reinstall.
2. One or more of the pods have issues around valid certificates.
    1. Diagnosis: The logs contains a message similar to the following `failed to verify certificate: x509: certificate signed by unknown authority`.
    2. Solution:
        1. This is usually due to an expired or invalid certificate in the cluster, and if so, renew the certificate.
        2. If the certificate is valid, but is signed by a local CA, make sure you have followed the procedure for a [local certificate authority](../../config/org-cert.md).
```

When installing a production cluster you may want to:

* Set one or more Run:ai system nodes. These are nodes dedicated to Run:ai software.

> * Machine learning frequently requires jobs that require CPU but __not GPU__. You may want to direct these jobs to dedicated nodes that do not have GPUs, so as not to overload these machines.

Reviewer: ```suggestion
* Machine learning frequently requires jobs that require CPU but **not GPU**. You may want to direct these jobs to dedicated nodes that do not have GPUs, so as not to overload these machines.
```
